### DISCUSSION TITLE
add error message for `skipToken` in `useSuspenseQueries`

### DISCUSSION THREAD

**Discussion by hautest** [TanStack/query#7768] [Ideas]
When using `skipToken` for `queryFn` in `useSuspenseQuery`, like in the example below:

```tsx
useSuspenseQuery({
  queryKey: ['useSuspenseQueryKey'],
  queryFn: skipToken,
})

```

The following specific error messages are displayed:

- ‘Attempted to invoke queryFn when set to skipToken. This is likely a configuration error. Query hash: '["suspenseQueryKey"]'’
- ‘skipToken is not allowed for useSuspenseQuery’

However, when using `skipToken` for `queryFn` in `useSuspenseQueries`, like in the example below:

```tsx
useSuspenseQueries({
  queries: [
    {
      queryKey: ['useSuspenseQueriesKey'],
      queryFn: skipToken,
    },
  ],
})

```

Only the following error message is displayed:

- ‘Attempted to invoke queryFn when set to skipToken. This is likely a configuration error. Query hash: '["useSuspenseQueriesKey"]’’

This makes it difficult to identify the specific error. What do you think about adding a more specific error message for `useSuspenseQueries` as well? [Here is a modification I have worked on.](https://github.com/hautest/tanstack-query/commit/483a6370913fb6f584dc6eb7d9a6fa5ef0126722)

 If this change looks good, can I go ahead and create a pull request?

### Before
<img width="498" alt="스크린샷 2024-07-20 오후 7 06 43" src="https://github.com/user-attachments/assets/c03338cb-01c2-416f-a2c8-879631217fbd">

### After

<img width="481" alt="스크린샷 2024-07-20 오후 6 55 58" src="https://github.com/user-attachments/assets/416881d7-e46c-49de-b2cb-9f76b61b2798">

**Comment by hautest** [comment]
Not only the useSuspenseQueries function but also the `useSuspenseInfiniteQuery` function does not provide detailed error messages regarding the inability to use skipToken.

**Comment by TkDodo** [comment]
looks good 👏

**Comment by hautest** [comment]
Thank you ! Here is the [Pull Request](https://github.com/TanStack/query/pull/7797)
